docs/topics/inline-classes.md

```diff
@@ -1,19 +1,29 @@
 [//]: # (title: Inline classes)
 
-> Inline classes are in [Alpha](#alpha-status-of-inline-classes). This feature may change incompatibly in the future and require manual migration. 
-> We will appreciate your feedback in [YouTrack](https://youtrack.jetbrains.com/issues/KT).
+> Inline classes are in [Beta](components-stability.md). It is almost stable, but migration steps may be required in the future. 
+> We'll do our best to minimize any changes you will have to make.
+> We appreciate your feedback on them in [YouTrack](https://youtrack.jetbrains.com/issue/KT-42434).
 >
 {type="warning"}
 
 Sometimes it is necessary for business logic to create a wrapper around some type. However, it introduces runtime 
 overhead due to additional heap allocations. Moreover, if the wrapped type is primitive, the performance hit is terrible, 
 because primitive types are usually heavily optimized by the runtime, while their wrappers don't get any special treatment. 
 
-To solve such issues, Kotlin introduces a special kind of class called an *inline class*, which is declared by placing an 
-`inline` modifier before the name of the class:
+To solve such issues, Kotlin introduces a special kind of class called an *inline class*. 
+Inline classes are a subset of value-based classes. They don't have identity and hold values only.
+
+To declare an inline class, use `inline` or `value` modifier before the name of the class.
+Kotlin also requires the `@JvmInline` annotation before class declaration for the JVM backend:
```

Comment:
The annotation is needed only before `value` classes, but the wording suggests, that one have to annotate `inline` classes as well.

Reply:
I've updated the description — wrote about `value` explicitly.